doc/source/complex.rst

```diff
@@ -0,0 +1,46 @@
+Complex Structures
```

Comment:
Maybe rename "Projects Using DyNet" and rename the file appropriately? The wording here seems a bit strange at the end of the documentation where the readers would presumably already know about DyNet's strong points. Other than that looks good.